include/simics/dmllib.h

```diff
@@ -766,11 +766,40 @@ _deserialize_object_trait_reference(ht_str_table_t *id_info_ht,
     return Sim_Set_Ok;
 }
 
-UNUSED __attribute__((const)) static inline bool
+UNUSED static void
+_signal_critical_error(const char *restrict format, ...) {
+    va_list va;
+    va_start(va, format);
+    char msg[512];
+    vsnprintf(msg, 512, format, va);
+    va_end(va);
+    VT_critical_error(msg, msg);
```

Comment:
This adds an ABI dependency requirement (user-built modules will require a recent Simics version), the reason why `_DML_fault` does its SIM_get_api_function dance is to avoid this.

This is OK (only breaks with base packages >2 years old), but please run a round of package tests to see if some compat test needs to be bumped. And if you conciously do this break, you should also clean up the DML_fault hack.

Reply:
This has already existed previously with `_memoized_recursion`, I just moved it up. If it didn't cause any issues then... :P

But sure, I can get rid of the `DML_fault` hack